src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts

```diff
@@ -266,14 +267,47 @@ export class MarkdownTextTransformer {
 	 * @param editorModel text model for the cell
 	 */
 	private isUndoOperation(selection: Selection, type: MarkdownButtonType, lineType: MarkdownLineType, editorModel: TextModel): boolean {
-		if (lineType === MarkdownLineType.BEGIN_AND_END_LINES) {
+		if (lineType === MarkdownLineType.BEGIN_AND_END_LINES || lineType === MarkdownLineType.WRAPPED_ABOVE_AND_BELOW) {
 			let selectedText = this.getExtendedSelectedText(selection, type, lineType, editorModel);
 			return selectedText && selectedText.startsWith(this.getStartTextToInsert(type)) && selectedText.endsWith(this.getEndTextToInsert(type));
 		} else {
 			return this.everyLineMatchesBeginString(selection, type, editorModel);
 		}
 	}
 
+	private handleUndoOperation(markdownLineType: MarkdownLineType, startRange: IRange, endRange: IRange, editorModel: TextModel, beginInsertedCode: string, endInsertedCode: string, selections: Selection[], selection: Selection): void {
+		if (markdownLineType === MarkdownLineType.BEGIN_AND_END_LINES) {
+			startRange = this.getIRangeWithOffsets(startRange, -1 * beginInsertedCode.length, 0, 0, 0);
+			endRange = this.getIRangeWithOffsets(endRange, 0, 0, endInsertedCode.length, 0);
+			editorModel.pushEditOperations(selections, [{ range: endRange, text: '' }, { range: startRange, text: '' }], null);
+		} else if (markdownLineType === MarkdownLineType.WRAPPED_ABOVE_AND_BELOW) {
+			startRange = this.getIRangeWithOffsets(startRange, 0, -1, 0, 0);
```

Comment:
Why is above & below only removing one character at the beginning/end? Shouldn't it be removing the entire begin/end InsertedCode? (so for example removing \`\`\` for code blocks)

Reply:
Oh nevermind - you're doing an entire row offset. Gotcha.